src/integrations.py

```diff
@@ -304,6 +308,31 @@ def update_oauth_client_config(self, ingress_url: str) -> None:
         self._requirer.update_client_config(client_config)
 
 
+@dataclass(frozen=True)
+class TLSCertificates:
+    ca_bundle: str
+
+    @classmethod
+    def load(cls, requirer: CertificateTransferRequires) -> "TLSCertificates":
+        """Fetch the CA certificates from all "receive-ca-cert" integrations.
+
+        Compose the trusted CA certificates in /etc/ssl/certs/ca-certificates.crt.
+        """
+        cert_transfer_integrations = requirer.charm.model.relations[
+            CERTIFICATE_TRANSFER_INTEGRATION_NAME
+        ]
+
+        ca_certs = {
+            integration.data[unit]["ca"]
+            for integration in cert_transfer_integrations
+            for unit in integration.units
+            if "ca" in integration.data[unit]
+        }
```

Comment:
question: Have you tested if the integration data is empty on relation broken (`certificate_removed`)?
If https://github.com/canonical/operator/issues/888 is still the case, you may end up pushing certs from a removed relation, because relation data can still be available on that event

Reply:
Yep, it is empty when the relation is broken.

I am a bit surprised by the behavior of the relation-broken event described in the issue. This design kind of beyond the user's expectation. Is this supposed to be that way or they have fixed it? (I am confused about what they finally decided)

Reply:
That's fine - in that case the certs lib is handling the removal then.
By design the data is gone only after relation-broken, I agree it's a little counter-intuitive. Issue is still opened, so I believe it hasn't changed.